lib/at_cmd_parser/Kconfig

```diff
@@ -6,4 +6,4 @@
 
 config AT_CMD_PARSER
 	bool "AT command parser library"
-
+	depends on NEWLIB_LIBC || EXTERNAL_LIBC
```

Comment:
Why do we need to depend on NEWLIB? Is it `strtoll()`?

Reply:
We should have `#ifdef`-ed the new functionality of this library that requires `newlibc`, and enabled that were necessary per-sample. As it is done right now, it is forcing a `libc` other than `minimal-libc` on all users of the parser, which is not ideal.